### PR TITLE
Implement createDatabase and dropDatabase on QueryInterface

### LINKED ISSUE #8821: support create / drop database with query interface
## What are you doing?
I'm trying to create a new schema with mysql using migrations

```js
module.exports = {
  up(queryInterface) {
    return queryInterface.createSchema('admin')
  },
  down(queryInterface) {
    return queryInterface.dropSchema('admin')
  },
}

```

## What do you expect to happen?
The schema to be created

## What is actually happening?
Schema not being created :cry: 

## Aditional info
Did a little research to understand what's happening and found this line https://github.com/sequelize/sequelize/blob/master/lib/dialects/mysql/query-generator.js#L19, which returns the `SHOW TABLES` SQL instead of a `CREATE DATABASE`

As a workaround, I could change the `queryInterface.createSchema` with `queryInterface.sequelize.query` and put my `CREATE DATABASE` query, which gives me exactly what I need. Just reporting a broken functionality (i guess it's broken)

If you need more info about it, just let me know :)

EDIT: Just to avoid confusion
> CREATE DATABASE creates a database with the given name. To use this statement, you need the CREATE privilege for the database. CREATE SCHEMA is a synonym for CREATE DATABASE.

https://dev.mysql.com/doc/refman/5.7/en/create-database.html

__Dialect:__ `mysql`
__Dialect version:__ `1.5.0`
__Database version:__ `8.0.3-rc-log`
__Sequelize version:__ `4.28.6`
__Tested with latest release:__ Yes, versions above

**Note :** _Your issue may be ignored OR closed by maintainers if it's not tested against latest version OR does not follow issue template._

### PR DESCRIPTION
**Pull request by ALiangLiang**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Resolved #8821

Added two API on QueryInterface - createDatabase and dropDatabase. I just want to used for https://github.com/sequelize/cli/issues/590. But these two API doesn't have testing code yet. It's hard to implement create and drop database in docker. If somebody has interesting on that. Please help me to add testing code on that.

### CODE REVIEW

**Comment by ALiangLiang** [comment]
I'm stuck on db testing. How can I create another testing database with db user 'sequelize_test'? I think it's not suitable to create and drop db 'sequelize_test'.

File: lib/dialects/abstract/query-generator.js [hunk 1]
```diff
@@ -964,6 +964,17 @@ class QueryGenerator {
     return table;
   }
 
+  /**
```
Comment (sushantdhiman) [inline lib/dialects/abstract/query-generator.js]: Remove this, use `quoteIdentifier` directly

File: lib/query-interface.js [hunk 2]
```diff
@@ -1303,6 +1331,10 @@ class QueryInterface {
     return this.QueryGenerator.quoteTable(identifier);
   }
 
+  quoteDatabase(identifier) {
```
Comment (sushantdhiman) [inline lib/query-interface.js]: Remove this as well

File: lib/dialects/mysql/query-generator.js [hunk 3]
```diff
@@ -16,6 +16,33 @@ class MySQLQueryGenerator extends AbstractQueryGenerator {
...
+    options = _.extend({
+      charset: null,
+      collate: null
+    }, options || {});
+
+    const query = 'CREATE DATABASE IF NOT EXISTS <%= database %><%= charset %><%= collation %>';
+
+    const values = {
+      database: this.quoteTable(databaseName),
+      charset: options.charset ? ' DEFAULT CHARSET SET = ' + options.charset : '',
```
Comment (sushantdhiman) [inline lib/dialects/mysql/query-generator.js]: Escape charset and collate

**Comment by ALiangLiang** [comment]
@sushantdhiman  I'm not familiar on MSSQL. So I can't validate the query string correctly. But I can add support and let you review the test code. Wait a moment.

**Comment by ALiangLiang** [comment]
BTW. Should I add support  on SQLite?

**Comment by sushantdhiman** [comment]
> BTW. Should I add support on SQLite?

No

**Comment by ALiangLiang** [comment]
> @ALiangLiang Please add support for MSSQL if possible, it should be similar to other dialects https://docs.microsoft.com/en-us/sql/t-sql/statements/create-database-transact-sql?view=sql-server-2017

done

**Comment by sushantdhiman** [comment]
Thanks @ALiangLiang

**Pull request merged**